cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -61,6 +64,31 @@ func GetStorageFactory(name string, host component.Host) (storage.Factory, error
 	return f, nil
 }
 
+// GetMetricsFactory locates the extension in Host and retrieves a metrics factory from it with the given name.
+func GetMetricsFactory(name string, host component.Host) (storage.MetricsFactory, error) {
+	var comp component.Component
+	for id, ext := range host.GetExtensions() {
+		if id.Type() == componentType {
+			comp = ext
+			break
+		}
+	}
+	if comp == nil {
+		return nil, fmt.Errorf(
+			"cannot find extension '%s' (make sure it's defined earlier in the config)",
+			componentType,
+		)
```

Comment:
the logic up to this point already exists in `GetStorageFactory`, let's move to a helper method `getExtension`